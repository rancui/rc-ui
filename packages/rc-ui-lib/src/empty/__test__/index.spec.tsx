import * as React from 'react';
import { cleanup, render } from '@testing-library/react';
import Button from '../../button';
import { Network } from '../Network';
import Empty from '..';

describe('Empty', () => {
  afterEach(() => {
    cleanup();
  });

  it('should render image correctly', () => {
    const { container } = render(<Empty image="search" />);
    expect(container).toMatchSnapshot();
  });

  it('should render description correctly', () => {
    const { container } = render(<Empty description="description" />);
    expect(container).toMatchSnapshot();
  });

  it('should render bottom correctly', () => {
    const { container } = render(
      <Empty>
        <Button round type="primary" className="bottom-button">
          按钮
        </Button>
      </Empty>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render svg when image is network component', () => {
    const { container } = render(<Empty image={Network} />);
    expect(container).toMatchSnapshot();
  });

  it('should render svg when prop image is network', () => {
    const { container } = render(<Empty image="network" />);
    expect(container).toMatchSnapshot();
  });

  it('should change image size when using image-size prop', async () => {
    const { container, rerender } = render(<Empty image="search" imageSize={50} />);

    const style = getComputedStyle(container.querySelector('.rc-empty__image'));

    expect(style.width).toEqual('50px');
    expect(style.height).toEqual('50px');

    rerender(<Empty image="search" imageSize="1vw" />);
    const style2 = getComputedStyle(container.querySelector('.rc-empty__image'));
    expect(style2.width).toEqual('1vw');
    expect(style2.height).toEqual('1vw');
  });
});
