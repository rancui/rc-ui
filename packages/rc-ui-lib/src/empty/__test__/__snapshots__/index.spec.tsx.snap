// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty should render bottom correctly 1`] = `
<div>
  <div
    class="rc-empty"
  >
    <div
      class="rc-empty__image"
    >
      <img
        alt=""
        src="https://img.yzcdn.cn/vant/empty-image-default.png"
      />
    </div>
    <div
      class="rc-empty__bottom"
    >
      <button
        class="bottom-button rc-button rc-button--primary rc-button--normal rc-button--round"
      >
        <div
          class="rc-button__content"
        >
          <span
            class="rc-button__text"
          >
            按钮
          </span>
        </div>
      </button>
    </div>
  </div>
</div>
`;

exports[`Empty should render description correctly 1`] = `
<div>
  <div
    class="rc-empty"
  >
    <div
      class="rc-empty__image"
    >
      <img
        alt=""
        src="https://img.yzcdn.cn/vant/empty-image-default.png"
      />
    </div>
    <p
      class="rc-empty__description"
    >
      description
    </p>
  </div>
</div>
`;

exports[`Empty should render image correctly 1`] = `
<div>
  <div
    class="rc-empty"
  >
    <div
      class="rc-empty__image"
    >
      <img
        alt=""
        src="https://img.yzcdn.cn/vant/empty-image-search.png"
      />
    </div>
  </div>
</div>
`;

exports[`Empty should render svg when image is network component 1`] = `
<div>
  <div
    class="rc-empty"
  >
    <div
      class="rc-empty__image"
    >
      <svg
        viewBox="0 0 160 160"
      >
        <defs>
          <lineargradient
            id="rc-empty-network-1"
            x1="64.022%"
            x2="64.022%"
            y1="100%"
            y2="0%"
          >
            <stop
              offset="0%"
              stop-color="#FFF"
              stop-opacity="0.5"
            />
            <stop
              offset="100%"
              stop-color="#F2F3F5"
            />
          </lineargradient>
          <lineargradient
            id="rc-empty-network-2"
            x1="50%"
            x2="50%"
            y1="0%"
            y2="84.459%"
          >
            <stop
              offset="0%"
              stop-color="#EBEDF0"
            />
            <stop
              offset="100%"
              stop-color="#DCDEE0"
              stop-opacity="0"
            />
          </lineargradient>
          <lineargradient
            id="rc-empty-network-3"
            x1="100%"
            x2="100%"
            y1="0%"
            y2="100%"
          >
            <stop
              offset="0%"
              stop-color="#EAEDF0"
            />
            <stop
              offset="100%"
              stop-color="#DCDEE0"
            />
          </lineargradient>
          <lineargradient
            id="rc-empty-network-4"
            x1="100%"
            x2="100%"
            y1="100%"
            y2="0%"
          >
            <stop
              offset="0%"
              stop-color="#EAEDF0"
            />
            <stop
              offset="100%"
              stop-color="#DCDEE0"
            />
          </lineargradient>
          <lineargradient
            id="rc-empty-network-5"
            x1="0%"
            x2="100%"
            y1="43.982%"
            y2="54.703%"
          >
            <stop
              offset="0%"
              stop-color="#EAEDF0"
            />
            <stop
              offset="100%"
              stop-color="#DCDEE0"
            />
          </lineargradient>
          <lineargradient
            id="rc-empty-network-6"
            x1="94.535%"
            x2="5.465%"
            y1="43.837%"
            y2="54.948%"
          >
            <stop
              offset="0%"
              stop-color="#EAEDF0"
            />
            <stop
              offset="100%"
              stop-color="#DCDEE0"
            />
          </lineargradient>
          <radialgradient
            cx="50%"
            cy="0%"
            fx="50%"
            fy="0%"
            gradientTransform="matrix(0 1 -.54835 0 .5 -.5)"
            id="rc-empty-network-7"
            r="100%"
          >
            <stop
              offset="0%"
              stop-color="#EBEDF0"
            />
            <stop
              offset="100%"
              stop-color="#FFF"
              stop-opacity="0"
            />
          </radialgradient>
        </defs>
        <g
          fill="none"
          fill-rule="evenodd"
        >
          <g
            opacity=".8"
          >
            <path
              d="M0 124V46h20v20h14v58H0z"
              fill="url(#rc-empty-network-1)"
              transform="matrix(-1 0 0 1 36 7)"
            />
            <path
              d="M121 8h22.231v14H152v77.37h-31V8z"
              fill="url(#rc-empty-network-1)"
              transform="translate(2 7)"
            />
          </g>
          <path
            d="M0 139h160v21H0z"
            fill="url(#rc-empty-network-7)"
          />
          <path
            d="M37 18a7 7 0 013 13.326v26.742c0 1.23-.997 2.227-2.227 2.227h-1.546A2.227 2.227 0 0134 58.068V31.326A7 7 0 0137 18z"
            fill="url(#rc-empty-network-2)"
            fill-rule="nonzero"
            transform="translate(43 36)"
          />
          <g
            opacity=".6"
            stroke-linecap="round"
            stroke-width="7"
          >
            <path
              d="M20.875 11.136a18.868 18.868 0 00-5.284 13.121c0 5.094 2.012 9.718 5.284 13.12"
              stroke="url(#rc-empty-network-3)"
              transform="translate(43 36)"
            />
            <path
              d="M9.849 0C3.756 6.225 0 14.747 0 24.146c0 9.398 3.756 17.92 9.849 24.145"
              stroke="url(#rc-empty-network-3)"
              transform="translate(43 36)"
            />
            <path
              d="M57.625 11.136a18.868 18.868 0 00-5.284 13.121c0 5.094 2.012 9.718 5.284 13.12"
              stroke="url(#rc-empty-network-4)"
              transform="rotate(-180 76.483 42.257)"
            />
            <path
              d="M73.216 0c-6.093 6.225-9.849 14.747-9.849 24.146 0 9.398 3.756 17.92 9.849 24.145"
              stroke="url(#rc-empty-network-4)"
              transform="rotate(-180 89.791 42.146)"
            />
          </g>
          <g
            fill-rule="nonzero"
            transform="translate(31 105)"
          >
            <rect
              fill="url(#rc-empty-network-5)"
              height="34"
              rx="2"
              width="98"
            />
            <rect
              fill="#FFF"
              height="18"
              rx="1.114"
              width="80"
              x="9"
              y="8"
            />
            <rect
              fill="url(#rc-empty-network-6)"
              height="6"
              rx="1.114"
              width="18"
              x="15"
              y="12"
            />
          </g>
        </g>
      </svg>
    </div>
  </div>
</div>
`;

exports[`Empty should render svg when prop image is network 1`] = `
<div>
  <div
    class="rc-empty"
  >
    <div
      class="rc-empty__image"
    >
      <svg
        viewBox="0 0 160 160"
      >
        <defs>
          <lineargradient
            id="rc-empty-network-1"
            x1="64.022%"
            x2="64.022%"
            y1="100%"
            y2="0%"
          >
            <stop
              offset="0%"
              stop-color="#FFF"
              stop-opacity="0.5"
            />
            <stop
              offset="100%"
              stop-color="#F2F3F5"
            />
          </lineargradient>
          <lineargradient
            id="rc-empty-network-2"
            x1="50%"
            x2="50%"
            y1="0%"
            y2="84.459%"
          >
            <stop
              offset="0%"
              stop-color="#EBEDF0"
            />
            <stop
              offset="100%"
              stop-color="#DCDEE0"
              stop-opacity="0"
            />
          </lineargradient>
          <lineargradient
            id="rc-empty-network-3"
            x1="100%"
            x2="100%"
            y1="0%"
            y2="100%"
          >
            <stop
              offset="0%"
              stop-color="#EAEDF0"
            />
            <stop
              offset="100%"
              stop-color="#DCDEE0"
            />
          </lineargradient>
          <lineargradient
            id="rc-empty-network-4"
            x1="100%"
            x2="100%"
            y1="100%"
            y2="0%"
          >
            <stop
              offset="0%"
              stop-color="#EAEDF0"
            />
            <stop
              offset="100%"
              stop-color="#DCDEE0"
            />
          </lineargradient>
          <lineargradient
            id="rc-empty-network-5"
            x1="0%"
            x2="100%"
            y1="43.982%"
            y2="54.703%"
          >
            <stop
              offset="0%"
              stop-color="#EAEDF0"
            />
            <stop
              offset="100%"
              stop-color="#DCDEE0"
            />
          </lineargradient>
          <lineargradient
            id="rc-empty-network-6"
            x1="94.535%"
            x2="5.465%"
            y1="43.837%"
            y2="54.948%"
          >
            <stop
              offset="0%"
              stop-color="#EAEDF0"
            />
            <stop
              offset="100%"
              stop-color="#DCDEE0"
            />
          </lineargradient>
          <radialgradient
            cx="50%"
            cy="0%"
            fx="50%"
            fy="0%"
            gradientTransform="matrix(0 1 -.54835 0 .5 -.5)"
            id="rc-empty-network-7"
            r="100%"
          >
            <stop
              offset="0%"
              stop-color="#EBEDF0"
            />
            <stop
              offset="100%"
              stop-color="#FFF"
              stop-opacity="0"
            />
          </radialgradient>
        </defs>
        <g
          fill="none"
          fill-rule="evenodd"
        >
          <g
            opacity=".8"
          >
            <path
              d="M0 124V46h20v20h14v58H0z"
              fill="url(#rc-empty-network-1)"
              transform="matrix(-1 0 0 1 36 7)"
            />
            <path
              d="M121 8h22.231v14H152v77.37h-31V8z"
              fill="url(#rc-empty-network-1)"
              transform="translate(2 7)"
            />
          </g>
          <path
            d="M0 139h160v21H0z"
            fill="url(#rc-empty-network-7)"
          />
          <path
            d="M37 18a7 7 0 013 13.326v26.742c0 1.23-.997 2.227-2.227 2.227h-1.546A2.227 2.227 0 0134 58.068V31.326A7 7 0 0137 18z"
            fill="url(#rc-empty-network-2)"
            fill-rule="nonzero"
            transform="translate(43 36)"
          />
          <g
            opacity=".6"
            stroke-linecap="round"
            stroke-width="7"
          >
            <path
              d="M20.875 11.136a18.868 18.868 0 00-5.284 13.121c0 5.094 2.012 9.718 5.284 13.12"
              stroke="url(#rc-empty-network-3)"
              transform="translate(43 36)"
            />
            <path
              d="M9.849 0C3.756 6.225 0 14.747 0 24.146c0 9.398 3.756 17.92 9.849 24.145"
              stroke="url(#rc-empty-network-3)"
              transform="translate(43 36)"
            />
            <path
              d="M57.625 11.136a18.868 18.868 0 00-5.284 13.121c0 5.094 2.012 9.718 5.284 13.12"
              stroke="url(#rc-empty-network-4)"
              transform="rotate(-180 76.483 42.257)"
            />
            <path
              d="M73.216 0c-6.093 6.225-9.849 14.747-9.849 24.146 0 9.398 3.756 17.92 9.849 24.145"
              stroke="url(#rc-empty-network-4)"
              transform="rotate(-180 89.791 42.146)"
            />
          </g>
          <g
            fill-rule="nonzero"
            transform="translate(31 105)"
          >
            <rect
              fill="url(#rc-empty-network-5)"
              height="34"
              rx="2"
              width="98"
            />
            <rect
              fill="#FFF"
              height="18"
              rx="1.114"
              width="80"
              x="9"
              y="8"
            />
            <rect
              fill="url(#rc-empty-network-6)"
              height="6"
              rx="1.114"
              width="18"
              x="15"
              y="12"
            />
          </g>
        </g>
      </svg>
    </div>
  </div>
</div>
`;
